### PR TITLE
updates for release

### PR DESCRIPTION
**Pull request by barrettk**
This is a more focused PR that includes changes made in https://github.com/metrumresearchgroup/pmparams/pull/66 and https://github.com/metrumresearchgroup/pmparams/pull/67 that we want get in before the next release.

Changes include:

### Core function changes
- `define_param_table()`:
   - Better documentation and examples
- `format_param_table()`:
   - New `.cleanup_cols` argument that automatically selects only the columns needed by `make_pmtable`. Deprecates `.select_cols`.
   - Better documentation and examples
- `make_pmtable()`:
   - Better documentation and examples
- `param_notes()`: 
   - Better documentation and examples
- Vignette updates
  - More clear wording in some cases (I didnt touch the paragraphs intentionally)
  - use `bbr` model as the example
  - Show tables in addition to printing the code

### Helper functions
- `formatGreekNames()`: minor update for use with `define_param_table()`
- `loadParamEstimates()`: Refactor the order in which we read in various forms of `.estimates` (e.g, `bbr` model vs `data.frame`) & ensure `bbr` is installed at various steps.
- `loadParamKey()`: minor update to have a more informative error message. Necessary for quality `define_param_table()` test.

### Utility
- `utils-pipe.R`: imports the `magrittr` pipe to be usable in function examples. This is consistent with our other packages.
- `utils.R` functions: used to prevent certain functions and tests from erroring out if `bbr` is not installed, as it is a suggested package
- Re-saves out `param_est.csv` to not include row names in the table: necessary for quality tests.
- CI and pkgr updates related to conflicts with the latest `glue` release (`1.8.0`). See [#67](https://github.com/metrumresearchgroup/pmparams/pull/67) for more details.

closes https://github.com/metrumresearchgroup/pmparams/issues/75, https://github.com/metrumresearchgroup/pmparams/issues/76

### CODE REVIEW

File: R/param-notes.R [hunk 1]
```diff
@@ -51,6 +59,7 @@ param_notes <- function(.ci = 95, .zscore = NULL){
     ci   = "CI: confidence intervals",
     corr = "Corr: correlation coefficient",
     cv   = "CV: coefficient of variation",
+    iqr = "IQR: interquartile range",
```
Comment (graceannobrien) [inline R/param-notes.R:62]: We don't want iqr addition to `param_notes()` in this version. We will only ever use it for bootstrap parameter tables, so it doesn't make sense to have in this version (iqr is not a supported CI in non-bootstrap functions). Keep example updates but drop iqr addtion.

**Comment by barrettk** [comment]
@graceannobrien I dont see any issues that this PR closes, but can you add "closes [issue #]" to the PR description if so? Im going to write up one for the CI issues and dependency updates to link to this for now.

File: R/format_param_table.R [hunk 2]
```diff
@@ -1,78 +1,136 @@
...
+#' @param .cleanup_cols logical (T/F). Defaults to `TRUE`, which selects the
+#'   following columns:
+#'   - `"type"`, `"abb"`, `"greek"`, `"desc"`, `"value"`, `"ci"`, `"shrinkage"`.
+#'   - Set to `FALSE` to return all columns.
+#' @param .prse logical (T/F). If `TRUE`, output `pRSE`. Default is `FALSE`.
+#' @param .digit set significant digits for output (optional). Default is three
+#'   digits.
+#' @param .maxex set maxex for computation (optional). Default is `NULL`.
+#' @param .select_cols Deprecated. Please use `.cleanup_cols` instead. Columns
+#'   to select for output. To return all columns, specify `"all"`.
```
Comment (graceannobrien) [inline R/format_param_table.R:42]: Remove 2nd and 3rd sentence. No longer relevant.

File: R/format_param_table.R [hunk 3]
```diff
@@ -1,78 +1,136 @@
...
   .df_out[[paste0("ci_", .ci_level)]] <-  .df_out$ci
+  .df_out <- .df_out %>% dplyr::select(-any_of(c("ci", "ci_level")))
 
-  if (any(tolower(.select_cols) == "all")) {
-    return(.df_out %>% dplyr::select(-ci) %>% as.data.frame())
+  # Handle deprecated .select_cols arg, or define it below
+  if (!is.null(.select_cols)){
+    message("`.select_cols` is deprecated and included for backwards compatibility. Using this sets `.cleanup_cols = FALSE`")
+    .cleanup_cols <- FALSE
+    # Check for deprecated ci column specification
```
[the remark this answers is not in the document]
Comment (barrettk) [reply R/format_param_table.R:108]: Discussed offline that this is not desired. Did lead to the creation of a new test though.

**Review by graceannobrien — APPROVED** [review]

**Review by graceannobrien — APPROVED** [review]

**Pull request merged**